addons/website_event/tests/test_website_event.py

```diff
@@ -129,3 +138,11 @@ def test_sitemap(self):
         self.assertTrue('/event/event-0' in resp.text, 'Published events must be present in the sitemap')
         self.assertTrue('/event/event-1' in resp.text, 'Published events must be present in the sitemap')
         self.assertFalse('/event/event-2' in resp.text, 'Unpublished events must not be present in the sitemap')
+
+    @users('user_portal')
+    def test_check_search_in_address(self):
+        ret = self.env['event.event']._search_get_detail(
+            self.website, order=None, options={'displayDescription':'', 'displayDetail':''}
+        )
+        result = ret['search_extra'](self.env, 'Turlock')[0][-1].get_result_ids()
+        self.assertEqual(*result, self.events[0].id, 'Event should exist for the searched term')
```

Comment:
can you do a self.assertRaises(AccessError) when trying to read the partner directly ?
(to make it explicit)